safe-global/web-core
# fix: show more errors in `SignOrExecuteForm`

## What it solves

Resolves #874

## How this PR fixes it

Errors are now shown for: wrong chain, non-proposal, sending to self, gas-related errors and submission errors.

## How to test it

- Connect to wrong chain and observe the "Your wallet is connected to the wrong chain." error.
- Try creating a transaction as a non-owner and observe the "You are currently not an owner of this Safe and won't be able to submit this transaction." error.
- Connect to a nested Safe and send funds to the parents. Observe the "Cannot execute a transaction from the Safe itself, please connect a different account." (This was present in `safe-react` but I am of the opinion that we should remove this, or extend it to all instances of owners?)
- Attempt to execute the `swapOwner` on `gno:0x9DF58F04F3e914ED8D972146401C85313A53FecA` and observe the `safeTxGas`-related error.

Note: gas-related errors change wording depending on if they are related to creation or execution:

- "This transaction will most likely fail. To save gas costs, avoid creating the transaction."
- "This transaction will most likely fail. To save gas costs, reject this transaction."

## Screenshots

![image](https://user-images.githubusercontent.com/20442784/195398397-9aa8fb31-fbf0-4e1d-8fd3-1cf8bb0968c8.png)

![image](https://user-images.githubusercontent.com/20442784/195398332-759eba54-7db6-4cd6-a224-3867724c5e14.png)
